src/server.js

```diff
@@ -286,10 +286,10 @@ function auth(data) {
 		}
 	} else {
 		client = manager.findClient(data.user, data.token);
-		var signedIn = data.token && data.token === client.config.token;
+		var signedIn = data.token && client && data.token === client.config.token;
```

Comment:
If you follow the rest of the code path in this case, when `findClient` returns false it treats the user as failing his authentication.
There is one place in `authFunction` (called by `authFunction(client, data.user, data.password, authCallback);`) which does make use of `client`, but it's wrapped around a `try`/`catch` so it's "safe".

I recognize that this is the cheapest solution to make this work though, do you have any suggestion to improve this?